src/ProfileListBuilder.php

```diff
@@ -134,19 +155,47 @@ public function buildRow(EntityInterface $entity) {
    * {@inheritdoc}
    */
   public function getOperations(EntityInterface $entity) {
+    /** @var \Drupal\profile\Entity\ProfileInterface $entity */
     $operations = parent::getOperations($entity);
 
     $destination = $this->redirectDestination->getAsArray();
     foreach ($operations as $key => $operation) {
       $operations[$key]['query'] = $destination;
     }
```

Comment:
Seems to me that this should be going after all operations are built so that the destination is added to all of them.